app/src/js/components/BatchAsyncCommands/BatchAsyncCommands.js

```diff
@@ -72,7 +72,8 @@ export class BatchCommand extends React.Component {
         selected,
         history,
         isOnModalConfirm: true,
-        isOnModalComplete: false
+        isOnModalComplete: false,
+        setState: this.setState.bind(this)
```

Comment:
I don't love the idea of passing `setState` like this. My preference would be to do something like creating this function and passing that to `getModalOptions`:

```
function handleModalClose () {
  this.setState({ activeModal: false });
}
```

Especially since setState controls the `BatchAsyncCommand` component as a whole rather than just the modal.

We can leave this as is, but it worries me a little.

Reply:
That's actually fine (and good) and I don't mind fixing it at all. I didn't like it either that much but justified it by thinking we might want to control other parts of the state, but we can do those similarly in the future.

Reply:
Cool, sounds good!

Reply:
e53a3b4a